pkg/monitor/controller.go

```diff
@@ -50,37 +77,65 @@ type controllerMonitor struct {
 func NewControllerMonitor(
 	client clientset.Interface,
 	nodeInformer coreinformers.NodeInformer,
+	externalNodeInformer externalnodeinformers.ExternalNodeInformer,
 	querier controllerquerier.ControllerQuerier,
 ) *controllerMonitor {
 	m := &controllerMonitor{
 		client:           client,
 		nodeInformer:     nodeInformer,
+		nodeLister:       nodeInformer.Lister(),
 		nodeListerSynced: nodeInformer.Informer().HasSynced,
+		nodeQueue:        workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(minRetryDelay, maxRetryDelay), "node"),
 		querier:          querier,
 		controllerCRD:    nil,
 	}
 	nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc:    nil,
+		AddFunc:    m.enqueueNode,
 		UpdateFunc: nil,
-		DeleteFunc: m.deleteStaleAgentCRD,
+		DeleteFunc: m.enqueueNode,
 	})
+	// Register Informer and add handlers for ExternalNode events only if the feature is enabled.
+	if features.DefaultFeatureGate.Enabled(features.ExternalNode) {
```

Comment:
Usually we do not directly use FeatureGate inside each module, but pass a boolean like "externalNodeEnabled" to a module from agent.go.

Reply:
Got it, I have passed the boolean as suggested.